registry/txt.go

```diff
@@ -331,32 +331,38 @@ func extractRecordType(name string) (baseName, recordType string) {
 	return name, ""
 }
 
-// dropAffix strips TXT record to find an endpoint name it manages
-// It takes into consideration a fact that it could contain record type
-// So it gets stripped first
-func (pr affixNameMapper) dropAffix(name string) string {
+// dropAffixExtractType strips TXT record to find an endpoint name it manages
+// it also returns the record type
+func (pr affixNameMapper) dropAffixExtractType(name string) (baseName, recordType string) {
 	if pr.recordTypeInAffix() {
 		for _, t := range getSupportedTypes() {
 			t = strings.ToLower(t)
 			iPrefix := strings.ReplaceAll(pr.prefix, recordTemplate, t)
 			iSuffix := strings.ReplaceAll(pr.suffix, recordTemplate, t)
+
 			if pr.isPrefix() && strings.HasPrefix(name, iPrefix) {
-				return strings.TrimPrefix(name, iPrefix)
+				return strings.TrimPrefix(name, iPrefix), t
 			}
 
 			if pr.isSuffix() && strings.HasSuffix(name, iSuffix) {
-				return strings.TrimSuffix(name, iSuffix)
+				return strings.TrimSuffix(name, iSuffix), t
 			}
 		}
+
+		// handle old TXT records
+		pr.prefix = pr.dropAffixTemplate(pr.prefix)
+		pr.suffix = pr.dropAffixTemplate(pr.suffix)
```

Comment:
This relies on `pr` being passed by value. We're one refactor away from incorrectly modifying the mapper. Perhaps extract to `prefix` and `suffix` local variables?